src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -1079,49 +1049,42 @@ private boolean endsWithVowelNoY(char s[], int len) {
   }
 
   /**
-   * Creates a CharArraySet from a file.
-   *
-   * @param stopwords
-   *          Input stream from the stopwords file
-   *
-   * @param matchVersion
-   *          the Lucene version for cross version compatibility
-   * @return a CharArraySet containing the distinct stopwords from the given
-   *         file
-   * @throws IOException
-   *           if loading the stopwords throws an {@link IOException}
-   */
-  private static CharArraySet loadStopwordSet(InputStream stopwords,
-      Version matchVersion) throws IOException {
-    Reader reader = null;
-    try {
-      reader = IOUtils.getDecodingReader(stopwords, IOUtils.CHARSET_UTF_8);
-      return WordlistLoader.getWordSet(reader, matchVersion);
-    } finally {
-      IOUtils.close(reader);
-    }
-  }
-
-  /**
    * Helper function loading an exception Map by its name from the config file.
    * @param exName The exceptions name
-   * @return a Map of "term" -> ['s','t','e','m','m','e','d']
+   * @return a CharArrayMap which maps terms to their exceptions
    * @throws FileNotFoundException
    */
-  private static Map<String, char[]> loadExceptions(String exName)
+  private CharArrayMap<char[]> loadExceptions(String exName)
           throws FileNotFoundException {
     InputStream in = SkroutzGreekStemmer.class.getClassLoader()
             .getResourceAsStream("stemmer.yml");
     Yaml reader = new Yaml();
 
     Map<String, Map<String, String>> composite = (Map) reader.load(in);
-    Map<String, String> stemException = composite.get(exName);
-    Map<String, char[]> rv = new HashMap<String, char[]>();
+    CharArrayMap<char[]> stemException = new CharArrayMap(1, true);
 
-    for (Map.Entry<String, String> entry : stemException.entrySet()){
-      rv.put(entry.getKey().toLowerCase(),
+    for (Map.Entry<String, String> entry : composite.get(exName).entrySet()){
+        stemException.put(entry.getKey().toLowerCase().toCharArray(),
               entry.getValue().toLowerCase().toCharArray());
     }
-    return rv;
+    return stemException;
+  }
+
+  /**
+   * TODO Consider if it's overall better to sacrifice modularity for
+   * efficiency: merge this with loadExceptions
+   * Helper function loading a list of protected words by its name from
+   * the config file.
+   * @param prName The protected words file name
+   * @return a CharArraySet of "term" -> ['s','t','e','m','m','e','d']
+   * @throws FileNotFoundException
+   */
+  private CharArraySet loadProtected(String prName)
+          throws FileNotFoundException {
```

Comment:
From my understanding, `getResourceAsStream` does not throw an exception, it just returns null.
Yaml reader load in the other hand throws an `java.io.IOException` if the file is not found.
I suggest changing the method signature to reflect that.